test/utils/daemonsets/daemonsets.go

```diff
@@ -0,0 +1,221 @@
+		if err != nil {
+			return fmt.Errorf("failed to get pods, err: %s", err)
+		}
+
+		if len(pods.Items) == 0 {
+			doneCleanUp = true
+			break
+		}
+		time.Sleep(5 * time.Second)
+	}
+
+	fmt.Printf("Successfully cleaned up daemon set %s\n", daemonSetName)
+	return nil
+}
+
+// Check if the daemon set exists
+func doesDaemonSetExist(daemonSetName, namespace string) bool {
+	_, err := client.Client.DaemonSets(namespace).Get(context.TODO(), daemonSetName, metav1.GetOptions{})
+	if err != nil {
+		fmt.Println("Error ocurred for" + err.Error())
+	}
+	// If the error is not found, that means the daemon set exists
+	return err == nil
+}
+
+func CreateDaemonSet(daemonSetName, namespace, containerName, imageWithVersion string, timeout time.Duration) (*v1core.PodList, error) {
+
+	rebootDaemonSet := CreateDaemonSetsTemplate(daemonSetName, namespace, containerName, imageWithVersion)
+
+	if doesDaemonSetExist(daemonSetName, namespace) {
+		err := DeleteDaemonSet(daemonSetName, namespace)
+		if err != nil {
+			logrus.Debug("Failed to delete L2discovery daemonset because: %s", err)
+		}
+	}
+
+	fmt.Printf("Creating daemon set %s\n", daemonSetName)
+	_, err := client.Client.DaemonSets(namespace).Create(context.TODO(), rebootDaemonSet, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+	WaitDaemonsetReady(namespace, daemonSetName, timeout)
+
+	fmt.Println("Deamon set is ready")
+
+	var ptpPods *v1core.PodList
+	ptpPods, err = client.Client.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: "name=" + daemonSetName})
+	if err != nil {
+		return ptpPods, err
+	}
+	fmt.Printf("Successfully created daemon set %s\n", daemonSetName)
+	return ptpPods, nil
+}
+
+func WaitDaemonsetReady(namespace, name string, timeout time.Duration) error {
+	oc := client.Client
+	nodes, err := oc.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get node list, err:%s", err)
+	}
+
+	nodesCount := int32(len(nodes.Items))
+	isReady := false
+	for start := time.Now(); !isReady && time.Since(start) < timeout; {
+		daemonSet, err := oc.AppsV1().DaemonSets(namespace).Get(context.Background(), name, metav1.GetOptions{})
+
+		if err != nil {
+			return fmt.Errorf("failed to get daemonset, err: %s", err)
+		}
+
+		if daemonSet.Status.DesiredNumberScheduled != nodesCount {
+			return fmt.Errorf("daemonset DesiredNumberScheduled not equal to number of nodes:%d, please instantiate debug pods on all nodes", nodesCount)
+		}
+
+		logrus.Infof("Waiting for (%d) debug pods to be ready: %+v", nodesCount, daemonSet.Status)
+		if isDaemonSetReady(&daemonSet.Status) {
+			isReady = true
+			break
+		}
+
+		time.Sleep(5 * time.Second)
+	}
+
+	if !isReady {
+		return errors.New("daemonset debug pods not ready")
+	}
+
+	logrus.Infof("All the debug pods are ready.")
+	return nil
+}
+
+func isDaemonSetReady(status *appsv1.DaemonSetStatus) bool {
```

Comment:
The unit tests for this function could have been copied over as well: https://github.com/test-network-function/cnf-certification-test/blob/main/pkg/provider/provider_test.go#L91